### ISSUE TITLE
Mask Input, Set Mask Dynamically

### ISSUE THREAD

**Issue by jjjrmy** [formkit/formkit#1086]
### Reproduction

https://formkit.link/fff14f502a47740aeadad4c9964586aa

### Describe the bug

https://formkit.link/57221193925e52224b5dd318e5da5243

It is not possible to dynamically set the mask for a Mask input

### Environment

Latest

**Comment by justin-schroeder** [comment]
This is not currently supported — but perhaps something to look into at some point in the future.

**Comment by jjjrmy** [comment]
@justin-schroeder 
I definitely understand that the FK Mask input is way more complex than a typical mask library.
From examples of these other libraries there are other cases for more than just a credit-card field that might require a dynamic mask.

Maybe it can be dynamic as long as you are not using select mode this should be possible?

**Comment by justin-schroeder** [comment]
Yes — and dynamic masks were considered at the inception of the mask input. We decided against it for non-technical reasons: we think it is almost always bad UX.

Having the expected value of an input change as you are interacting with it is confusing and also really challenging to make accessible in an appropriate way. That said, there are, of course, several pretty valid use cases where having a dynamic mask is just a slight formatting tweak and doesn’t fundamentally change how the input expects values — so in some limited use cases I think it would be appropriate for us to support it.

**Comment by jjjrmy** [comment]
Exactly! It's more of the actual Mask / Placeholder that is changing than the actual expected value.
But I can definitely see it being problematic if someone were trying to change `###` mask to `aaa`
